### PR TITLE
fix problem with duplicate vehicle Ids

### PR DESCRIPTION
**Pull request by rewertvsp**
fix problem with duplicate vehicle Ids while the creation of vehicles for the carriers

### CODE REVIEW

File: contribs/application/src/main/java/org/matsim/freightDemandGeneration/CarrierReaderFromCSV.java [hunk 1]
```diff
@@ -463,20 +457,21 @@ static void createNewCarrierAndAddVehilceTypes(Scenario scenario,
 				carriers.addCarrier(thisCarrier);
 			}
 			if (singleNewCarrier.getVehicleDepots() == null)
-				singleNewCarrier.setVehicleDepots(new String[] {});
-			while (singleNewCarrier.getVehicleDepots().length < singleNewCarrier.getNumberOfDepotsPerType()) {
-				Random rand = new Random();
+				singleNewCarrier.setVehicleDepots(new ArrayList<String>());
+			Random rand = new Random(4711);
```
Comment (michalmac) [inline contribs/application/src/main/java/org/matsim/freightDemandGeneration/CarrierReaderFromCSV.java:461]: Maybe initialise `rand` with a seed like `singleNewCarrier.name.hashCode()`. Otherwise you will always get the same link sequence for all carriers, which may not be desired if you want to have them more spread. `singleNewCarrier.name.hashCode()` makes it also reproducible

File: contribs/application/src/main/java/org/matsim/freightDemandGeneration/CarrierReaderFromCSV.java [hunk 2]
```diff
@@ -485,10 +480,18 @@ static void createNewCarrierAndAddVehilceTypes(Scenario scenario,
...
 								thisType.getId().toString() + "_" + thisCarrier.getId().toString() + "_" + singleDepot
 										+ "_start" + singleNewCarrier.getVehicleStartTime() + "_" + (i + 1),
-								Vehicle.class), Id.createLinkId(singleDepot), thisType)
+								Vehicle.class);
+						while (carrierCapabilities.getCarrierVehicles().containsKey(vehilcelId)) {
+							vehilcelId = Id.create(thisType.getId().toString() + "_" + thisCarrier.getId().toString()
+									+ "_" + singleDepot + "_V" + countDepots + "_start"
+									+ singleNewCarrier.getVehicleStartTime() + "_" + (i + 1), Vehicle.class);
+							countDepots++;
+						}
```
Comment (michalmac) [inline contribs/application/src/main/java/org/matsim/freightDemandGeneration/CarrierReaderFromCSV.java:492]: Since now the vehicle depots are unique links, you can revert this change (starting from line 474).

**Review by michalmac — APPROVED** [review]
Thanks for fixing it. Looks good to me. I left 2 comments that you may consider.

File: contribs/application/src/main/java/org/matsim/freightDemandGeneration/CarrierReaderFromCSV.java (hunk 1 above)
Comment (rewertvsp) [reply contribs/application/src/main/java/org/matsim/freightDemandGeneration/CarrierReaderFromCSV.java]: Thanks for this idea. I will change it

File: contribs/application/src/main/java/org/matsim/freightDemandGeneration/CarrierReaderFromCSV.java (hunk 2 above)
Comment (rewertvsp) [reply contribs/application/src/main/java/org/matsim/freightDemandGeneration/CarrierReaderFromCSV.java:496]: Now I added the possibility to add a depot on an existing depot location, if no other possible link is found. Thats why I do not revert this change
